src/Security/SslContext.cs

```diff
@@ -535,9 +569,27 @@ public static IntPtr GetTypeId ()
 		// TODO: Headers say /* Deprecated, does nothing */ but we are not completly sure about it since there is no deprecation macro
 		// Plus they added new members to SslSessionStrengthPolicy enum opened radar://23379052 https://trello.com/c/NbdTLVD3
 		// Xcode 8 beta 1: the P/Invoke was removed completely.
+
+#if !NET
 		[Unavailable (PlatformName.iOS, message : "'SetSessionStrengthPolicy' is not available anymore.")]
 		[Unavailable (PlatformName.MacOSX, message : "'SetSessionStrengthPolicy' is not available anymore.")]
+#else
+		[UnsupportedOSPlatform ("ios")]
+		[UnsupportedOSPlatform ("macos")]
+#endif
+#if !NET
 		[Obsolete ("'SetSessionStrengthPolicy' is not available anymore.")]
+#else
+#if __MACCATALYST__
+		[Obsolete ("Starting with maccatalyst 'SetSessionStrengthPolicy' is not available anymore.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
```

Comment:
There's no version here, so this message ends up weird.

Maybe just skip the "Starting with ..." part?

```suggestion
		[Obsolete ("'SetSessionStrengthPolicy' is not available anymore.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
```

and the same for the other ones below.